androidApp/src/main/java/com/mbta/tid/mbta_app/android/state/subscribeToPredictions.kt

```diff
@@ -88,6 +88,7 @@ class PredictionsViewModel(
 
     fun disconnect() {
         predictionsRepository.disconnect()
+        _predictions.value = null
```

Comment:
Resetting the predictions on disconnect prevents showing stale data while reconnecting - I'm surprised we didn't have this in place before.

Reply:
Was this presenting issues as is for your changes? Looking at the iOS implementation, [leavePredictions](https://github.com/mbta/mobile_app/blob/5108bc6e9520e10055c4ee1f8a19bc20837a6b31/iosApp/iosApp/Pages/NearbyTransit/NearbyTransitView.swift#L270) only disconnects from the channel. 

I thought we wanted to keep predictions around for the case where a user backgrounds the app / disconnects from the channel briefly, then returns and should still see the latest predictions.

Reply:
Yes; the existing behavior was causing "Predictions unavailable" to display while the predictions channel reconnects, because as is we don't [reset the predictions](https://github.com/mbta/mobile_app/blob/5108bc6e9520e10055c4ee1f8a19bc20837a6b31/iosApp/iosApp/Pages/NearbyTransit/NearbyTransitView.swift#L213) when the location gets reset. I can scope this a little more narrowly to avoid that issue without breaking the background behavior, though.

Reply:
>    // reset predictions if nearby stop list has been reset

I'm struggling a bit with this logic. Couldn't the stop list be reset to a new list rather than null?

Reply:
`NearbyTransitViewModel.reset` sets `nearby = null`, which means that the `NearbyTransitView`'s `stopIds` will be `null`. Do you find 27c34a54 clearer?

Reply:
Yes, thank you!